spring-integration-ws/src/main/java/org/springframework/integration/ws/dsl/MarshallingWsInboundGatewaySpec.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ws.dsl;
+
+import org.springframework.integration.ws.MarshallingWebServiceInboundGateway;
+import org.springframework.oxm.Marshaller;
+import org.springframework.oxm.Unmarshaller;
+
+/**
+ * The spec for a {@link MarshallingWebServiceInboundGateway}.
+ *
+ * @author Gary Russell
+ * @since 5.3
+ *
+ */
+public class MarshallingWsInboundGatewaySpec extends BaseWsInboundGatewaySpec<MarshallingWsInboundGatewaySpec,
+	MarshallingWebServiceInboundGateway> {
+
+	protected MarshallingWsInboundGatewaySpec() {
+		super(new MarshallingWebServiceInboundGateway());
+	}
+
+	/**
+	 * Construct an instance with the provided {@link Marshaller} (which must also implement
+	 * {@link Unmarshaller}).
+	 * @param marshaller the marshaller.
+	 */
+	protected MarshallingWsInboundGatewaySpec(Marshaller marshaller) {
+		super(new MarshallingWebServiceInboundGateway(marshaller));
+	}
+
+	/**
+	 * Construct an instance with the provided arguments.
+	 * @param marshaller the marshaller.
+	 * @param unmarshaller the unmarshaller.
+	 */
+	protected MarshallingWsInboundGatewaySpec(Marshaller marshaller, Unmarshaller unmarshaller) {
```

Comment:
I wouldn't expose two of these ctors since it feels much natural just to call those configuration options...